otter/convergence/selfheal.py

```diff
@@ -0,0 +1,193 @@
+"""
+Self heal service.
+"""
+
+from effect import Effect
+from effect.do import do, do_return
+
+from kazoo.exceptions import LockTimeout
+from kazoo.protocol.states import KazooState
+
+from toolz.curried import filter
+
+from twisted.application.internet import TimerService
+from twisted.application.service import MultiService
+from twisted.internet.defer import inlineCallbacks, returnValue
+
+from txeffect import perform
+
+from otter.convergence.composition import tenant_is_enabled
+from otter.convergence.service import trigger_convergence
+from otter.log.intents import with_log
+from otter.models.intents import GetAllValidGroups, GetScalingGroupInfo
+from otter.models.interface import ScalingGroupStatus
+from otter.util.zk import GetChildren
+
+
+class SelfHeal(MultiService, object):
+    """
+    A service that triggers convergence on all the groups on interval basis.
+    Only one node is allowed to do this.
+
+    :ivar disp: Effect dispatcher to perform all effects (ZK, CASS, log, etc)
+    :type disp: Either :obj:`ComposedDispatcher` or :obj:`TypeDispatcher`
+
+    :ivar log: Twisted logger with .msg and .err methods
+    :ivar lock: Lock object from :obj:`TxKazooClient`
+    :ivar :obj:`TxKazooClient` kz_client: Twistified kazoo client object
```

Comment:
This part of it is documented here: http://www.sphinx-doc.org/en/stable/domains.html#python-signatures